STATISTICS.md

```diff
@@ -0,0 +1,523 @@
+# Statistics
+
+librdkafka may be configured to emit internal metrics at a fixed interval
+by setting the `statistics.interval.ms` configuration property to a value > 0
+and registering a `stats_cb` (or similar, depending on language).
+
+The stats are provided as a JSON object string.
+
+**Note**: The metrics returned may not be completely consistent between
+          brokers, toppars and totals, due to the internal asynchronous
+          nature of librdkafka.
+          E.g., the top level `tx` total may be less than the sum of
+          the broker `tx` values which it represents.
+
+
+## General structure
+
+All fields that contain sizes are are in bytes unless otherwise noted.
+
+```
+{
+ <Top-level fields>
+ "brokers": {
+    <brokers fields>,
+    "toppars": { <toppars fields> }
+ },
+ "topics": {
+   <topic fields>,
+   "partitions": {
+     <partitions fields>
+   }
+ }
+[, "cgrp": { <cgrp fields> } ]
+}
+```
+
+## Field type
+
+Fields are represented as follows:
+ * string - UTF8 string.
+ * int - Integer counter (64 bits wide). Ever increasing.
+ * int gauge - Integer gauge (64 bits wide). Will be reset to 0 on each stats emit.
+ * object - Nested JSON object.
+ * bool - `true` or `false`.
+
+
+## Top-level
+
+Field | Type | Example | Description
+----- | ---- | ------- | -----------
+name | string | `"rdkafka#producer-1"` | Handle instance name
+client_id | string | `"rdkafka"` | The configured (or default) `client.id`
+type | string | `"producer"` | Instance type (producer or consumer)
+ts | int | 12345678912345 | librdkafka's internal monotonic clock (micro seconds)
+time | int | | Wall clock time in seconds since the epoch
+replyq | int gauge | | Number of ops waiting in queue for application to serve with rd_kafka_poll()
```

Comment:
... for the application ...
it would be nice to have a quick indication here of what constitutes an 'op'.